src/mlpack/methods/softmax_regression/softmax_regression.hpp

```diff
@@ -57,10 +57,34 @@ namespace regression {
 
 template<
   template<typename> class OptimizerType = mlpack::optimization::L_BFGS
->
+  >
 class SoftmaxRegression
 {
  public:
+  /**
+   * Initialize the SoftmaxRegression without performing training.
+   * Default value of lambda is 0.0001.
+   * Be sure to use Train() before calling Predict() or ComputeAccuracy(),
+   * otherwise the results may be meaningless.
+   *
+   * @param inputSize Size of the input feature vector.
+   * @param numClasses Number of classes for classification.
+   * @param fitIntercept add intercept term or not.
+   */
+  SoftmaxRegression(const size_t inputSize,
+                    const size_t numClasses,
+                    const bool fitIntercept = false);
+
+  /**
+   * Construct the SoftmaxRegression class with the provided data and labels.
+   * This will train the model.
+   *
+   * @param fileName name of the files saving the model contents
+   * @param name name of the structure to be save
+   * @exception If the file cannot be load, the exception will thrown
+   */
+  SoftmaxRegression(const std::string &fileName,
+                    const std::string& name);
```

Comment:
I think we should avoid this constructor, because no other mlpack methods provide a constructor of this sort.  If we implemented this method here, then we'd have to implement it anywhere (for consistency).  I'd prefer that users use `data::Load()` instead (maybe we can revisit this later, but for now I don't have time to create that constructor for every mlpack method before the next release).